src/main.rs

```diff
@@ -3,50 +3,47 @@ mod parameters;
 mod population_loader;
 mod transmission_manager;
 
-use clap::Parser;
+use clap::Args;
+use ixa::runner::run_with_custom_args;
 use ixa::{
-    context::Context, error::IxaError, global_properties::ContextGlobalPropertiesExt,
-    people::ContextPeopleExt, random::ContextRandomExt, report::ContextReportExt,
+    Context, ContextGlobalPropertiesExt, ContextPeopleExt, ContextRandomExt, ContextReportExt,
+    IxaError,
 };
 use std::path::PathBuf;
 use transmission_manager::InfectiousStatus;
 
 use crate::parameters::Parameters;
 use crate::population_loader::{Age, CensusTract};
 
-#[derive(Parser)]
-#[command(version, about, long_about = None)]
-struct Args {
-    /// path to the input file
-    #[arg(short, long)]
-    input_file: PathBuf,
-
-    /// path to the output directory
-    #[arg(short, long)]
-    output_directory: PathBuf,
-
+#[derive(Args, Debug)]
+struct CustomArgs {
     /// whether force overwrite of output files if they already exist
-    #[arg(short, long, default_value = "false", default_missing_value = "true")]
+    #[arg(short = 'f', long)]
     force_overwrite: bool,
 }
 
-fn initialize(args: &Args) -> Result<Context, IxaError> {
-    let mut context = Context::new();
-    // Read the global properties.
-    context.load_global_properties(&args.input_file)?;
+fn initialize() -> Result<Context, IxaError> {
+    let mut context = run_with_custom_args(|context, args, custom_args: Option<CustomArgs>| {
+        println!("Setting Overwrite parameter");
```

Comment:
I assume this was for debugging purposes. I would remove this print statement now and not have it on main.